client/packages/common/src/hooks/useUrlQuery/useUrlQueryParams.ts

```diff
@@ -81,26 +84,95 @@ export const useUrlQueryParams = ({
     updateQuery({ [key]: value });
   };
 
-  const filterKeyArray: string[] = !filterKey
-    ? []
-    : typeof filterKey === 'string'
-    ? [filterKey]
-    : filterKey;
+  const getFilterBy = () => {
+    const filterBy: FilterBy = {};
 
-  const filter: FilterController = {
-    onChangeStringFilterRule: (key: string, _, value: string) =>
-      updateFilterQuery(key, value),
-    onChangeDateFilterRule: () => {},
-    onClearFilterRule: key => updateFilterQuery(key, ''),
-    filterBy: filterKeyArray.reduce<FilterBy>((prev, key) => {
+    const filterKeyArray: string[] = !filterKey
+      ? []
+      : typeof filterKey === 'string'
+      ? [filterKey]
+      : filterKey;
+
+    filterKeyArray.reduce<FilterBy>((prev, key) => {
       const queryValue = urlQuery[key];
       if (!queryValue) return prev;
 
-      prev[key] = {
+      filterBy[key] = {
         [filterCondition]: String(queryValue),
       };
       return prev;
     }, {}),
+      additionalFilters.forEach(filter => {
+        if (urlQuery[filter.key]) {
+          let filterValue = urlQuery[filter.key];
+          switch (filterValue) {
+            case 'true':
+              filterValue = true;
+              break;
+            case 'false':
+              filterValue = false;
+              break;
+          }
+          if (filter.condition === 'between') {
+            if (filter.key) {
+              const filterItems = String(urlQuery[filter.key]).split('_');
+              const dateAfter = filterItems[0]
```

Comment:
Allow null so that the api doesn't throw an error if one of the range params is missing..